lib/acme/resources/challenges/dvsni01.rb

```diff
@@ -0,0 +1,12 @@
+class Acme::Resources::Challenges::DVSNI01 < Acme::Resources::Challenges::Base
+
+  def sni_name
+    hd = crypto.digest.hexdigest(authorization_key)
+    "#{hd[0..31]}.#{hd[32..64]}.acme.invalid"
+  end
+
+  def request_verification
+    response = client.connection.post(@uri, { resource: 'challenge', type: 'tls-dvsni-01', keyAuthorization: authorization_key })
```

Comment:
Well, I'm testing against the acme-staging server and that seems to work. It seems it doesn't check the "type" field at all and does the validation still against the self-signedSNI certificate?!